Added big sample size button

MBrinks2239 opened the pull request:
Feature where there is a button to allow a bigger sample size, adjustable by the sliders as per your suggestion.

I also added translations files this time, however all non-English languages were translated by AI.
Also added a button to more easily clear the large sample size.

This PR replaces #126

MBrinks2239 commented:
Also I am looking into implementing #101 to improve UI performance and maybe adding a loader while its simulating.
This could be merged for now and this be added in one of my following PR's. Whatever you prefer.

MBrinks2239 commented:
Fixed issue sliders not working

MBrinks2239 commented:
Implemented #101 now too, let me know if you don't want me to and remove the commit.

Frostiae commented:
> @Frostiae can you check out the PR? Let me know if you need anything changed.

Yup I will check it out once I have some time, should be able to this week.

MBrinks2239 commented:
1. When adjusting the slider it does jump back to the smaller sample size. This was intentional because otherwise the live updates would be very laggy and take a long time. I do agree that replacing these two buttons with a single checkbox could be a nice change.

2. I actually tested it on knight using an LC sword where skills indeed do work. The issue seems to happen when either waterbomb or counter attack is added.

The issue seems to have to do with these lines:
```js
    let data = {};

     // THIS LINE HERE     V
    // Add waterbomb
    if (Context.settings.waterbombEnabled && Context.attacker.getStat("skillchance", true, 11389) > 0) {
        Context.player.skillLevels[11389] = 1;
    }
    
     // AND THIS LINE HERE     V
    // Counter attack
    if (Context.player.skillLevels[2506] != undefined) {
        Context.player.skillLevels[6725] = Context.player.skillLevels[2506];
    }

    for (const [skill, level] of Object.entries(Context.player.skillLevels)) {
        if (level <= 0) {
            continue; // Shouldn't happen
        }
        console.log("Skill: " + skill);

        const skillProp = Utils.getSkillById(skill);
        const levelProp = skillProp.levels[level - 1];
  ``` 
  
  Could you explain to me the purpose of these lines? The waterbomb one I understand, as it simple adds waterbomb to the player's active list of skills. However, what exactly you are doing with counter-attack I am not quite sure. When checking the skills file, I can see one of them is called `Atk. Counter` and the other is called `Counter attack`. This would help me understand the use of the line and where the issue might be located, or what I should move it if it's not usable inside the web worker.

MBrinks2239 commented:
Alright I fixed the issue causing skills to not load properly, apparently the editing of the static context wasn't available in the worker thread since it wasn't on the same thread, thus the editing of the skills did not work since the edits were not saved correctly.

I also changed the two buttons with a single checkbox, like you requested. I did however not use a disabled property on them, as a user might want to change these values before trigging the long loading sequence that would take place when triggering the larger sample size. Also, like I said before, when users start editing the values that change how big the bigger sample size is, it does automatically revert to the default sample size to avoid a laggy user experience.

If you need any further changes, let me know!

MBrinks2239 commented:
@Frostiae could you check out the final PR again?

MBrinks2239 commented:
@Frostiae any update on this?
